### ISSUE TITLE
Update unit constants to new format

### ISSUE THREAD

**Issue by phidauex** [neggert/hass-egauge#315]
## Describe the bug

New versions of HomeAssistant are introducing new unit constants, and current unit constants are planned for deprecation in version 2025.1 (a ways out, for sure). Current operation is not impacted.

Full log below, but the constants that need to be changed are:

- ELECTRIC_POTENTIAL_VOLT -> UnitOfElectricPotential.VOLT
- ENERGY_KILO_WATT_HOUR -> UnitOfEnergy.KILO_WATT_HOUR
- POWER_WATT -> UnitOfPower.WATT 
- PRESSURE_PA -> UnitOfPressure.PA 
- TEMP_CELSIUS -> UnitOfTemperature.CELSIUS
- STATE_CLASS_MEASUREMENT -> SensorStateClass.MEASUREMENT
- STATE_CLASS_TOTAL_INCREASING -> SensorStateClass.TOTAL_INCREASING

## Debug log

```text
ELECTRIC_POTENTIAL_VOLT was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfElectricPotential.VOLT instead, please create a bug report at https://github.com/neggert/egauge/issues
ENERGY_KILO_WATT_HOUR was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/neggert/egauge/issues
POWER_WATT was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPower.WATT instead, please create a bug report at https://github.com/neggert/egauge/issues
PRESSURE_PA was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPressure.PA instead, please create a bug report at https://github.com/neggert/egauge/issues
TEMP_CELSIUS was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please create a bug report at https://github.com/neggert/egauge/issues
```

```
STATE_CLASS_MEASUREMENT was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use SensorStateClass.MEASUREMENT instead, please create a bug report at https://github.com/neggert/egauge/issues
STATE_CLASS_TOTAL_INCREASING was used from egauge, this is a deprecated constant which will be removed in HA Core 2025.1. Use SensorStateClass.TOTAL_INCREASING instead, please create a bug report at https://github.com/neggert/egauge/issues
```

**Comment by phidauex** [comment]
Here is a commit to another integration showing a similar change with the new constant and an updated import line: https://github.com/Limych/ha-average/commit/7bcdbdde2cdd5ca7c5be71a6d6a3d9a695c42abc

**Comment by pewhite22** [comment]
Is there any plans to update this integration? It was working nicely for me before. I'd update it myself, but I'm dumb :D

**Comment by homewirelessguy** [comment]
This is an awesome integration! Please consider updating it to solve the #315 issue above. Thank you.

**Comment by billraff** [comment]
Would also request this be updated if possible before the new year. Awesome integration.

**Comment by jdwhite** [comment]
@neggert - 2025.1 is under 2 months away now and this issue and #317 have items that need updating before 2025.1  

If you need assistance, please reach out to the community. 

Can we please get an acknowledgement or sign of life that this integration is still active? Thanks.

**Comment by neggert** [comment]
Hey, sorry all. I don't check in here very often anymore. I'll take a look when I have time. PRs are also welcome if someone else gets a chance to look at it.

**Comment by neggert** [comment]
This should be resolved with the 0.5.0 release. Let me know if you're still seeing any issues after updating.
